### PR TITLE
fix: routegroup does not apply default algorithm

### PR DESCRIPTION
**Pull request by szuecs**
fix: hard coded default algorithm that should not be set in this part of the code
test: add test cases

### CODE REVIEW

File: dataclients/kubernetes/testdata/routegroups/loadbalancer-algorithm/default-ch-algorithm-with-override.kube [hunk 1]
```diff
@@ -0,0 +1 @@
+default-lb-algorithm: consistantHash
```
Comment (MustafaSaber) [inline dataclients/kubernetes/testdata/routegroups/loadbalancer-algorithm/default-ch-algorithm-with-override.kube:1]: wrong spelling again `consistentHash`

Comment (szuecs) [reply dataclients/kubernetes/testdata/routegroups/loadbalancer-algorithm/default-ch-algorithm-with-override.kube]: fixed

**Comment by MustafaSaber** [comment]
👍

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**